### DISCUSSION TITLE
Guidance on Enabling Background Audio Playback

### DISCUSSION THREAD

**Discussion by MarkIvanDev** [unoplatform/uno#10129] [Q&A]
I would like to ask if there is guidance on enabling background audio playback in Uno apps? From what I understand, in UWP, it is just a simple enabling background media playback in the manifest. In Android, this is usually implemented by foreground services, although by using them, the player must be moved from using the MediaPlayerElement to an internal MediaPlayer inside the foreground service. In iOS, adding the background control must be added to the AppDelegate before iOS 13, and SceneDelegate from iOS 13 and overriding RemoteControlReceived in UIViewController. It just seems that all three platforms implement this functionality very differently from one another and might involve implementing unique playback logic for each platform head.

**Comment by jeromelaban** [comment]
At this time, Uno does not provide unified background audio playback and we haven't had lots of requests for it at this time.

You are right that for now, using platform specific APIs directly is the best way to go at this time, using the available resources and documentation for Xamarin apps.

**Comment by BrianDT** [comment]
@MarkIvanDev The Plugin.Maui.Audio does a good job on Windows, Android and iOS
